examples/asr/speech_to_text.py

```diff
@@ -62,16 +55,86 @@
     +model.optim.args.betas=[0.8,0.5]\
     +model.optim.args.weight_decay=0.0005
 
+Finetune a model
+1) Finetune from a .nemo file
+
+```sh
+    python examples/asr/speech_to_text.py \
+        --config-path=<path to dir of configs> \
+        --config-name=<name of config without .yaml>) \
+        model.train_ds.manifest_filepath="<path to manifest file>" \
+        model.validation_ds.manifest_filepath="<path to manifest file>" \
+        trainer.gpus=-1 \
+        trainer.max_epochs=50 \
+        +init_from_nemo_model="<path to .nemo model file>"
+```
+
+2) Finetune from a pretrained model (via NGC)
+
+```sh
+    python examples/asr/speech_to_text.py \
+        --config-path=<path to dir of configs> \
+        --config-name=<name of config without .yaml>) \
+        model.train_ds.manifest_filepath="<path to manifest file>" \
+        model.validation_ds.manifest_filepath="<path to manifest file>" \
+        trainer.gpus=-1 \
+        trainer.max_epochs=50 \
+        +init_from_pretrained_model="<name of pretrained checkpoint>"        
+```
 """
 
+import pytorch_lightning as pl
+from omegaconf import OmegaConf, open_dict
+
+from nemo.collections.asr.models import EncDecCTCModel
+from nemo.core.config import hydra_runner
+from nemo.utils import logging
+from nemo.utils.exp_manager import exp_manager
+
+
+def restore_weights_if_required(model: EncDecCTCModel, cfg: OmegaConf):
```

Comment:
This function runs every time. What happens in multiple runs to continue training?
It is going to download and load the weights every time. Let's add to check if if it is the first step otherwise ignore this weight loading.